File: question/leetcode/leetcode1265/leetcode1265.go
package leetcode

/*   Below is the interface for ImmutableListNode, which is already defined for you.
 *
 *   type ImmutableListNode struct {
 *
 *   }
 *
 *   func (this *ImmutableListNode) getNext() ImmutableListNode {
 *		// return the next node.
 *   }
 *
 *   func (this *ImmutableListNode) printValue() {
 *		// print the value of this node.
 *   }
 */

// TODO: not implemented
type ImmutableListNode struct {
	Val int
}

func (this *ImmutableListNode) getNext() ImmutableListNode {
	return ImmutableListNode{}
}

func (this *ImmutableListNode) printValue() {
}

// func printLinkedListInReverse(head ImmutableListNode) {
// 	if head != nil {
// 		printLinkedListInReverse(head.getNext())
// 		head.printValue()
// 	}
// }

File: question/leetcode/leetcode1265/leetcode1265.ts
/**
 * // This is the ImmutableListNode's API interface.
 * // You should not implement it, or speculate about its implementation
 * class ImmutableListNode {
 *      printValue() {}
 *
 *      getNext(): ImmutableListNode {}
 * }
 */
class ImmutableListNode {
	printValue() { }

	getNext(): ImmutableListNode { }
}


function printLinkedListInReverse(head: ImmutableListNode) {
	if (head != null) {
		printLinkedListInReverse(head.getNext());
		head.printValue();
	}
};